### PR TITLE
[WIP]: Move raid system to globalevents

### PR DESCRIPTION
**Pull request by liontts**
"This effectively nukes the raid system in favor of global events, which permit better management for raids. It's easier to define raids on interval, on exact times, force execution, force scheduling, add monsters on different delays and randomized amounts and positions... and also gets rid of a command.

I added a lib with helper functions to add monsters to a raid. The event is called onRaid and there is an example bundled pretty equivalent to the previous XML raid example."

Credits to @ranisalt, code taken from the tfs repository.

I didn't have time to generate the raids, I hope someone takes the trouble to do it.

### CODE REVIEW

**Comment by ranisalt** [comment]
Thanks for the mention :)

**Comment by gpedro** [comment]
can you remove tools folder and the cmake changes?

**Comment by liontts** [comment]
@gpedro This has already been done, I just need help fixing the error while compiling

**Comment by gpedro** [comment]
@ranisalt can you help us? xd

File: src/luascript.cpp [hunk 1]
```diff
@@ -4848,25 +4848,18 @@ int LuaScriptInterface::luaGameCreateMonsterType(lua_State* L)
...
-		return 1;
-	}
-
-	if (g_game.raids.getRunning()) {
-		lua_pushnumber(L, RETURNVALUE_ANOTHERRAIDISALREADYEXECUTING);
-		return 1;
+	auto&& eventMap = g_globalEvents->getEventMap(GLOBALEVENT_TIMER);
+	try {
+		GlobalEvent* event = eventMap.at(eventName);
+		lua_pushboolean(L, event->executeEvent());
```
Comment (ranisalt) [inline src/luascript.cpp]: Try
```cpp
GlobalEvent& event = eventMap.at(eventName);
lua_pushboolean(L, event.executeEvent());
```

**Review by ranisalt — CHANGES_REQUESTED** [review]

**Pull request closed without merging**